<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="statek_name" default="$(eval open(find('statek_config') + '/name').read())"/>

    <!-- Launch lidar -->
    <group ns="$(arg statek_name)/laser/">
        <node name="ydlidar_node" pkg="ydlidar_ros" type="ydlidar_node" output="screen" respawn="true">
            <param name="port" type="string" value="/dev/ydlidar"/>
            <param name="baudrate" type="int" value="128000"/>
            <param name="frame_id" type="string" value="$(arg statek_name)/laser/laser_link"/>
            <param name="resolution_fixed" type="bool" value="true"/>
            <param name="auto_reconnect" type="bool" value="true"/>
            <param name="reversion" type="bool" value="false"/>
            <param name="angle_min" type="double" value="-180" />
            <param name="angle_max" type="double" value="180" />
            <param name="range_min" type="double" value="0.1" />
            <param name="range_max" type="double" value="12.0" />
            <param name="ignore_array" type="string" value="" />
            <param name="frequency" type="double" value="8"/>
            <param name="samp_rate" type="int" value="5"/>
        </node>
    </group>

</launch>
